### PR TITLE
refactor(example,ml): android embedding v2

### PR DESCRIPTION
**Pull request by panmari**
## Description

Migrated ML vision example app away from deprecated android embeddings. Process is described at
https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects

Tested by running

    flutter run

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Comment by Salakar** [comment]
@Panmari would you mind providing a screenshot of the app running after this change? Thanks

**Comment by panmari** [comment]
Sure, here's a screenshot:

image

File: packages/firebase_ml_vision/example/pubspec.yaml [hunk 1]
```diff
@@ -26,7 +26,6 @@ dependency_overrides:
     path: ../../firebase_core/firebase_core_web
 
 dev_dependencies:
-  e2e: ^0.6.1
```
Comment (Salakar) [inline packages/firebase_ml_vision/example/pubspec.yaml:29]: This change isn't a complete removal of e2e, there's still usages here:  https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_ml_vision/example/android/app/src/androidTest/java/io/flutter/plugins/firebasemlvision/EmbeddingV1ActivityTest.java

**Comment by Salakar** [comment]
> Sure, here's a screenshot:
> 

Thanks

File: packages/firebase_ml_vision/example/pubspec.yaml (hunk 1 above)
Comment (panmari) [reply packages/firebase_ml_vision/example/pubspec.yaml]: Thanks for pointing me there! Reverted the change.

I naively ran 
```
flutter drive test_driver/firebase_ml_vision.dart
```
and it worked without e2e, so I assumed it's not necessary. But that's better left to another PR anyway....

**Review by Salakar — APPROVED** [review]

**Comment by panmari** [comment]
Anything else you need from my side?

**Comment by Salakar** [comment]
> Anything else you need from my side?

All good, thanks for the PR 👍

**Pull request merged**